library/std/src/io/error.rs

```diff
@@ -156,6 +191,62 @@ pub enum ErrorKind {
     /// [`Ok(0)`]: Ok
     #[stable(feature = "rust1", since = "1.0.0")]
     WriteZero,
+    /// The underlying storage (typically, a filesystem) is full.
+    ///
+    /// This does not include out of quota errors.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    NoStorageSpace,
+    /// Invalid seek
+    ///
+    /// Seeking was attempted on an open file handle which is not suitable for seeking - for
+    /// example, on Unix, a named pipe opened with `File::new`.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    InvalidSeek,
+    /// Filesystem quota was exceeded.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    FilesystemQuotaExceeded,
+    /// File larger than allowed or supported.
+    ///
+    /// This might arise from a hard limit of the underlying filesystem or file access API, or from
+    /// an administratively imposed resource limitation.  Simple disk full, and out of quota, have
+    /// their own errors.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    FileTooLarge,
+    /// Resource is busy.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    ResourceBusy,
+    /// Executable file is busy.
+    ///
+    /// An attempt was made to write to a file which is also in use as a running program.  (Not all
+    /// operating systems detect this situation.)
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    ExecutableFileBusy,
+    /// Deadlock (avoided)
+    ///
+    /// A file locking operation would result in deadlock.  This situation is typically detected, if
+    /// at all, on a best-effort basis.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    Deadlock,
+    /// Cross-device (hard) link.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    CrossDeviceLink,
+    /// Too many (hard) links to the same filesystem object.
+    ///
+    /// The filesystem does not support making so many hardlinks to the same file.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    TooManyLinks,
```

Comment:
This looks like window's [`ERROR_TOO_MANY_LINKS`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--1000-1299-)